lib/createStylelint.js

```diff
@@ -16,43 +16,29 @@ const STOP_DIR = IS_TEST ? process.cwd() : undefined;
 /**
  * The stylelint "internal API" is passed among functions
  * so that methods on a stylelint instance can invoke
- * each other while sharing options and caches
+ * each other while sharing options and caches.
+ *
  * @param {import('stylelint').StylelintStandaloneOptions} options
  * @returns {StylelintInternalApi}
  */
 module.exports = function createStylelint(options = {}) {
-	/** @type {Partial<StylelintInternalApi>} */
+	/** @type {StylelintInternalApi} */
+	// @ts-expect-error -- TS2740: Type '{ _options: StylelintStandaloneOptions; }' is missing the following properties from type 'StylelintInternalApi'
 	const stylelint = { _options: options };
 
-	// @ts-ignore TODO TYPES found out which cosmiconfig types are valid
-	stylelint._extendExplorer = cosmiconfig(null, {
-		transform: augmentConfig.augmentConfigExtended.bind(
-			null,
-			/** @type {StylelintInternalApi} */ (stylelint),
-		),
+	stylelint._extendExplorer = cosmiconfig('', {
```

Comment:
[note] The first argument of `cosmiconfig()` must be a string:

https://github.com/davidtheclark/cosmiconfig/blob/v7.0.1/src/index.ts#L110-L121